api/v1beta1/heat_webhook.go

```diff
@@ -99,10 +100,55 @@ var _ webhook.Validator = &Heat{}
 func (r *Heat) ValidateCreate() (admission.Warnings, error) {
 	heatlog.Info("validate create", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object creation.
+	var allErrs field.ErrorList
+	basePath := field.NewPath("spec")
+	if err := r.Spec.ValidateCreate(basePath); err != nil {
+		allErrs = append(allErrs, err...)
+	}
+
+	if len(allErrs) != 0 {
+		return nil, apierrors.NewInvalid(
+			schema.GroupKind{Group: "ironic.openstack.org", Kind: "Ironic"},
```

Comment:
Should this be `heat` and `Heat` instead of `ironic` and `Ironic`?

Reply:
right, good catch, thanks